### PR TITLE
#5665 PR builds using Jenkinsfile

### PR DESCRIPTION
**Pull request by AdamBurstynski**
#### What does this PR do?

Moved Codeco.io to be right after the ITests.

Removed all the commented windows sections.

Removed unneeded parallel sections as well.

Added in Commenting onto PR builds using internal library function postCommentIfPR.

#### Select relevant component teams: 
@codice/build 
@codice/continuous-integration 

#### Ask 2 committers to review/merge the PR and tag them here.
@TonyMorrison 
@shaundmorris 
@LinkMJB 

#### How should this be tested?

PR build should pass

#### What are the relevant tickets?
Fixes: #5665

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by AdamBurstynski** [comment]
build now

**Comment by AdamBurstynski** [comment]
build now

File: Jenkinsfile [hunk 1]
```diff
@@ -196,35 +158,28 @@ pipeline {
                 }
             }
         }
-        stage('Codecov.io upload') {
+
+        stage ('SonarCloud') {
             when {
-                expression {env.CHANGE_ID == null}
+                // Sonar Cloud only supports a single branch in the free/OSS tier
```
Comment (shaundmorris) [inline Jenkinsfile]: Thanks for adding the comment here.

**Comment by AdamBurstynski** [comment]
build now

**Pull request merged**